core/src/main/java/org/apache/druid/math/expr/ExprListenerImpl.java

```diff
@@ -188,9 +195,15 @@ public void exitLogicalAndOrExpr(ExprParser.LogicalAndOrExprContext ctx)
   @Override
   public void exitLongArray(ExprParser.LongArrayContext ctx)
   {
-    Long[] values = new Long[ctx.LONG().size()];
+    Long[] values = ctx.longArrayBody() == null
+                    ? new Long[0]
+                    : new Long[ctx.longArrayBody().longElement().size()];
     for (int i = 0; i < values.length; i++) {
-      values[i] = Long.parseLong(ctx.LONG(i).getText());
+      if (ctx.longArrayBody().longElement(i).getText().equalsIgnoreCase(Expr.NULL_LITERAL)) {
+        values[i] = null;
+      } else {
+        values[i] = Long.parseLong(ctx.longArrayBody().longElement(i).getText());
```

Comment:
Wondering if the array elements should be casted if they are not longs.

Reply:
Yeah, it seems more druidish to do that, I've modified explicitly typed numeric arrays to use a more permissive parsing with a couple of methods I've added to `Numbers` utility class, so `<LONG>[1, 1.2, 3]` and `<DOUBLE>[1, 2, 3]` will coerce the elements as `new Long[]{1L, 1L, 3L} ` and `new Double[]{1.0, 2.0, 3.0}` respectively, as one might hope. However, I am not converting string literals, so things like `<LONG>[1, 2, '3']` are still not valid at this time.

I also made explicit string arrays more permissive, and can accept any type of literal as an element and will convert them all to strings, `<STRING>['hello', 1, 1.2]` -> `new String[]{"hello", "1", "1.2"}`.

Reply:
Thanks, sounds good.